<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Arthur Wuster" />

<meta name="date" content="2016-07-19" />

<title>geneAttribution: Identification of Candidate Genes Associated with Noncoding Genetic Variation</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; }
code > span.dt { color: #902000; }
code > span.dv { color: #40a070; }
code > span.bn { color: #40a070; }
code > span.fl { color: #40a070; }
code > span.ch { color: #4070a0; }
code > span.st { color: #4070a0; }
code > span.co { color: #60a0b0; font-style: italic; }
code > span.ot { color: #007020; }
code > span.al { color: #ff0000; font-weight: bold; }
code > span.fu { color: #06287e; }
code > span.er { color: #ff0000; font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<link href="data:text/css,body%20%7B%0A%20%20background%2Dcolor%3A%20%23fff%3B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20max%2Dwidth%3A%20700px%3B%0A%20%20overflow%3A%20visible%3B%0A%20%20padding%2Dleft%3A%202em%3B%0A%20%20padding%2Dright%3A%202em%3B%0A%20%20font%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0A%20%20font%2Dsize%3A%2014px%3B%0A%20%20line%2Dheight%3A%201%2E35%3B%0A%7D%0A%0A%23header%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0A%0A%23TOC%20%7B%0A%20%20clear%3A%20both%3B%0A%20%20margin%3A%200%200%2010px%2010px%3B%0A%20%20padding%3A%204px%3B%0A%20%20width%3A%20400px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20border%2Dradius%3A%205px%3B%0A%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20font%2Dsize%3A%2013px%3B%0A%20%20line%2Dheight%3A%201%2E3%3B%0A%7D%0A%20%20%23TOC%20%2Etoctitle%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%20%20font%2Dsize%3A%2015px%3B%0A%20%20%20%20margin%2Dleft%3A%205px%3B%0A%20%20%7D%0A%0A%20%20%23TOC%20ul%20%7B%0A%20%20%20%20padding%2Dleft%3A%2040px%3B%0A%20%20%20%20margin%2Dleft%3A%20%2D1%2E5em%3B%0A%20%20%20%20margin%2Dtop%3A%205px%3B%0A%20%20%20%20margin%2Dbottom%3A%205px%3B%0A%20%20%7D%0A%20%20%23TOC%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dleft%3A%20%2D2em%3B%0A%20%20%7D%0A%20%20%23TOC%20li%20%7B%0A%20%20%20%20line%2Dheight%3A%2016px%3B%0A%20%20%7D%0A%0Atable%20%7B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dcolor%3A%20%23DDDDDD%3B%0A%20%20border%2Dstyle%3A%20outset%3B%0A%20%20border%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0A%20%20border%2Dwidth%3A%202px%3B%0A%20%20padding%3A%205px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%20%20line%2Dheight%3A%2018px%3B%0A%20%20padding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0A%20%20border%2Dleft%2Dstyle%3A%20none%3B%0A%20%20border%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Ap%20%7B%0A%20%20margin%3A%200%2E5em%200%3B%0A%7D%0A%0Ablockquote%20%7B%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20padding%3A%200%2E25em%200%2E75em%3B%0A%7D%0A%0Ahr%20%7B%0A%20%20border%2Dstyle%3A%20solid%3B%0A%20%20border%3A%20none%3B%0A%20%20border%2Dtop%3A%201px%20solid%20%23777%3B%0A%20%20margin%3A%2028px%200%3B%0A%7D%0A%0Adl%20%7B%0A%20%20margin%2Dleft%3A%200%3B%0A%7D%0A%20%20dl%20dd%20%7B%0A%20%20%20%20margin%2Dbottom%3A%2013px%3B%0A%20%20%20%20margin%2Dleft%3A%2013px%3B%0A%20%20%7D%0A%20%20dl%20dt%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%7D%0A%0Aul%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%7D%0A%20%20ul%20li%20%7B%0A%20%20%20%20list%2Dstyle%3A%20circle%20outside%3B%0A%20%20%7D%0A%20%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dbottom%3A%200%3B%0A%20%20%7D%0A%0Apre%2C%20code%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20color%3A%20%23333%3B%0A%7D%0Apre%20%7B%0A%20%20white%2Dspace%3A%20pre%2Dwrap%3B%20%20%20%20%2F%2A%20Wrap%20long%20lines%20%2A%2F%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20margin%3A%205px%200px%2010px%200px%3B%0A%20%20padding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Acode%20%7B%0A%20%20font%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0A%20%20font%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0A%20%20padding%3A%202px%200px%3B%0A%7D%0A%0Adiv%2Efigure%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0A%20%20background%2Dcolor%3A%20%23FFFFFF%3B%0A%20%20padding%3A%202px%3B%0A%20%20border%3A%201px%20solid%20%23DDDDDD%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20margin%3A%200%205px%3B%0A%7D%0A%0Ah1%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%20%20font%2Dsize%3A%2035px%3B%0A%20%20line%2Dheight%3A%2040px%3B%0A%7D%0A%0Ah2%20%7B%0A%20%20border%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20padding%2Dbottom%3A%202px%3B%0A%20%20font%2Dsize%3A%20145%25%3B%0A%7D%0A%0Ah3%20%7B%0A%20%20border%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20font%2Dsize%3A%20120%25%3B%0A%7D%0A%0Ah4%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0A%20%20margin%2Dleft%3A%208px%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Ah5%2C%20h6%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23ccc%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Aa%20%7B%0A%20%20color%3A%20%230033dd%3B%0A%20%20text%2Ddecoration%3A%20none%3B%0A%7D%0A%20%20a%3Ahover%20%7B%0A%20%20%20%20color%3A%20%236666ff%3B%20%7D%0A%20%20a%3Avisited%20%7B%0A%20%20%20%20color%3A%20%23800080%3B%20%7D%0A%20%20a%3Avisited%3Ahover%20%7B%0A%20%20%20%20color%3A%20%23BB00BB%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%0A%2F%2A%20Class%20described%20in%20https%3A%2F%2Fbenjeffrey%2Ecom%2Fposts%2Fpandoc%2Dsyntax%2Dhighlighting%2Dcss%0A%20%20%20Colours%20from%20https%3A%2F%2Fgist%2Egithub%2Ecom%2Frobsimmons%2F1172277%20%2A%2F%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Keyword%20%2A%2F%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%2F%2A%20DataType%20%2A%2F%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%2F%2A%20DecVal%20%28decimal%20values%29%20%2A%2F%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20BaseN%20%2A%2F%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Float%20%2A%2F%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Char%20%2A%2F%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20String%20%2A%2F%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%2F%2A%20Comment%20%2A%2F%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%2F%2A%20OtherToken%20%2A%2F%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20AlertToken%20%2A%2F%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Function%20calls%20%2A%2F%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%2F%2A%20ErrorTok%20%2A%2F%0A%0A" rel="stylesheet" type="text/css" />

</head>

<body>



<div id="header">
<h1 class="title">geneAttribution: Identification of Candidate Genes Associated with Noncoding Genetic Variation</h1>
<h4 class="author"><em>Arthur Wuster</em></h4>
<h4 class="date"><em>2016-07-19</em></h4>
</div>


<p><code>geneAttribution</code> is an R package for identifying the most likely gene or genes through which variation at a given genomic locus in the human genome acts. A typical use case is the annotation of results from genome-wide association studies (GWAS). The majority of variants identified by GWAS are located in noncoding regions and likely act by affecting gene expression (<a href="http://doi.org/10.1126/science.1222794">Maurano et al. 2012</a>). Variants typically contain multiple genes in the region of linkage disequilibrium and identifying the causative one is challenging.</p>
<p>The most basic functionality of <code>geneAttribution</code> assumes that the closer gene is to the input locus, the more likely the gene is to be causative. Additionally, any empirical data that links genomic regions to genes (e.g. expression quantitative trait loci (eQTL) or genome conformation data) can be used if it is supplied in the <a href="https://genome.ucsc.edu/FAQ/FAQformat.html">UCSC .BED file</a> format.</p>
<div id="basic-functionality" class="section level2">
<h2>Basic functionality</h2>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Basic workflow, default parameters</span>
geneLocs &lt;-<span class="st"> </span><span class="kw">geneModels</span>() <span class="co"># Define gene models</span>
<span class="kw">ga</span>(<span class="st">&quot;chr2&quot;</span>, <span class="dv">127156000</span>, geneLocs) <span class="co"># Get candidate gene probabilities</span>

<span class="co"># Basic workflow, customized parameters</span>
<span class="kw">library</span>(TxDb.Hsapiens.UCSC.hg19.knownGene) <span class="co"># Load TxDB object containing gene models</span>
geneLocs &lt;-<span class="st"> </span><span class="kw">geneModels</span>(<span class="dt">txdb=</span>TxDb.Hsapiens.UCSC.hg19.knownGene, <span class="dt">maxGeneLength=</span><span class="fl">1e5</span>) <span class="co"># Define gene models</span>
<span class="kw">ga</span>(<span class="st">&quot;chr2&quot;</span>, <span class="dv">127156000</span>, geneLocs, <span class="dt">lambda=</span><span class="fl">1.0e-5</span>, <span class="dt">maxDist=</span><span class="fl">2e6</span>, <span class="dt">minPP=</span><span class="dv">0</span>) <span class="co"># Get candidate gene probabilities</span></code></pre>
<p>The most basic functionality assumes that the closer a gene is to the input locus, the more likely it is to be a candidate gene. The relationship between distance to locus and candidate gene likelihood is modeled as an exponential distribution. The likelihoods for each gene are then normalized by dividing by the sum of likelihoods. As a result, the presence of multiple genes in the vicinity of a locus decreases the posterior probabilities of the individual genes.</p>
<div id="genemodels" class="section level3">
<h3>geneModels()</h3>
<p>To calculate the distance between the input locus and genes, gene models are required. For this, <code>geneAttribution</code> provides the <code>geneModels()</code> function. <code>geneModels()</code> takes a TxDb object containing genomic coordinates of genes as input and returns gene models in GenomicRanges format, with gene names in the <code>symbol</code> column. Loading the gene models may take a few minutes. The default for the TxDb input is <code>TxDb.Hsapiens.UCSC.hg38.knownGene</code>, which contains gene models for genome build GRCh38. An alternative input would be <code>TxDb.Hsapiens.UCSC.hg19.knownGene</code>, which contains gene models for build hg19. The geneModels function has additional optional input:</p>
<ul>
<li><code>maxGeneLength</code>. Gene models that are longer than this are excluded</li>
<li><code>genesToInclude</code> and <code>genesToExclude</code>. A character vector of gene symbols of genes to include (e.g. only protein coding genes) or exclude</li>
</ul>
</div>
<div id="ga" class="section level3">
<h3>ga()</h3>
<p>The minimum required input for <code>ga()</code> is a chromosome identifier in the same format than in the gene models and a chromosomal position in the same build than the gene models. The <code>ga()</code> function has additional optional input:</p>
<ul>
<li><code>lambda</code>, the <span class="math">\(\lambda\)</span> parameter for the exponential distribution modeling the candidate gene likelihood based on an exponential distribution. The default, based on empirical eQTL data from the Genome-Tissue Expression Project (<a href="http://doi.org/10.1126/science.1262110">GTEx Consortium 2015</a>), is 7.61e-06. Decreasing <code>lambda</code> gives genes that are close to the input locus a higher probability and decreasing it gives genes that are further away a higher probability</li>
<li><code>maxDist</code>, the maximum distance from the input locus at which a gene will be considered. Genes that are more than <code>maxDist</code> bases away from the input locus will be ignored. The default is 1,000,000 bases</li>
<li><code>minPP</code>, the minimum posterior probability at which a gene will be reported. Genes whose posterior probability is less than <code>minPP</code> will be summarized as “Other”. Set this to 0 to report all genes</li>
</ul>
</div>
</div>
<div id="using-empirical-data" class="section level2">
<h2>Using empirical data</h2>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Typical workflow, using sample data supplied with geneAttribution</span>
geneLocs &lt;-<span class="st"> </span><span class="kw">geneModels</span>()
fileName1 &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;hiCRegions.b38.bed&quot;</span>, <span class="dt">package=</span><span class="st">&quot;geneAttribution&quot;</span>)
fileName2 &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;eqtlHaplotypeBlocks.b38.bed&quot;</span>, <span class="dt">package=</span><span class="st">&quot;geneAttribution&quot;</span>)
empirical &lt;-<span class="st"> </span><span class="kw">loadBed</span>(<span class="kw">c</span>(fileName1, fileName2), <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">5</span>))
<span class="kw">ga</span>(<span class="st">&quot;chr2&quot;</span>, <span class="dv">127156000</span>, geneLocs, empirical)

<span class="co"># As above, but with user-supplied empirical data file in UCSC .BED format</span>
geneLocs &lt;-<span class="st"> </span><span class="kw">geneModels</span>()
empirical &lt;-<span class="st"> </span><span class="kw">loadBed</span>(<span class="st">&quot;INPUT_FILE.bed&quot;</span>, <span class="dt">weights=</span><span class="fl">1.5</span>) <span class="co"># INPUT_FILE.bed is a correctly formatted .BED file</span>
<span class="kw">ga</span>(<span class="st">&quot;chr2&quot;</span>, <span class="dv">127156000</span>, geneLocs, empirical)</code></pre>
<p>In addition, <code>geneAttribution</code> can make use of empirical data that links genomic loci to genes. eQTLs, which link genetic variants to the expression of specific genes, are an example of this. If the input locus is located inside the regions defined in the empirical data, the likelihood of the associated gene is multiplied by the associated weighting. User-provided empirical data can be loaded by using the <code>loadBed()</code> function.</p>
<p>The loadBed funtion reads user-provided files in UCSC .BED format. .BED files must be tab-separated and have the following column headers: <code>chr start end symbol</code>. An optional fifth <code>score</code> column may also be provided. Gene symbols used in the empirical data must match the symbols used for the gene models and the genome build (e.g. GRCh38) must also match the genome build of the gene models.</p>
<p>Together with the .BED files, weightings for the data may be provided. The default for this is 2, which doubles the likelihoods for genes if the input locus is located in a region defined by the empirical data. A weighting of 1 would not change the likelihood, and a weighting of less than 1 would decrease the likelihood. Alternatively to reading .BED files, users may construct empirical data themselves as a list of GenomicRanges objects with a score column containing the weights.</p>
<p>The <code>extdata</code> directory provides two .BED files in genome build GRCh38:</p>
<ul>
<li><code>hiCRegions.b38.bed</code>, which contains Capture Hi-C genome conformation data linking promoters to other genomic regions in the GM12878 CD34 cell line (<a href="http://doi.org/10.1038/ng.3286">Mifsud et al. 2015</a>)</li>
<li><code>eqtlHaplotypeBlocks.b38.bed</code>, which defines haplotype blocks with eQTLs in at least two different tissues from the Genome-Tissue Expression Project (<a href="http://doi.org/10.1126/science.1262110">GTEx Consortium 2015</a>)</li>
</ul>
<p>Because both files are meant as samples, they are limited to chromosome 2.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
